txnkv/txnlock/lock_resolver.go

```diff
@@ -291,120 +294,111 @@ func (lr *LockResolver) BatchResolveLocks(bo *retry.Backoffer, locks []*Lock, lo
 //    commit status.
 // 3) Send `ResolveLock` cmd to the lock's region to resolve all locks belong to
 //    the same transaction.
-func (lr *LockResolver) ResolveLocks(bo *retry.Backoffer, callerStartTS uint64, locks []*Lock) (int64, []uint64 /*pushed*/, error) {
-	return lr.resolveLocks(bo, callerStartTS, locks, false, false)
+func (lr *LockResolver) ResolveLocks(bo *retry.Backoffer, callerStartTS uint64, locks []*Lock) (int64, error) {
+	ttl, _, _, err := lr.resolveLocks(bo, callerStartTS, locks, false, false)
+	return ttl, err
 }
 
-// ResolveLocksLite resolves locks while preventing scan whole region.
-func (lr *LockResolver) ResolveLocksLite(bo *retry.Backoffer, callerStartTS uint64, locks []*Lock) (int64, []uint64 /*pushed*/, error) {
-	return lr.resolveLocks(bo, callerStartTS, locks, false, true)
+// ResolveLocksForRead is essentially the same as ResolveLocks, except with some optimizations for read.
+// Read operations needn't wait for resolve secondary locks and can read through(the lock's transaction is committed
+// and its commitTS is less than or equal to callerStartTS) or ignore(the lock's transaction is rolled back or its minCommitTS is pushed) the lock .
+func (lr *LockResolver) ResolveLocksForRead(bo *retry.Backoffer, callerStartTS uint64, locks []*Lock, lite bool) (int64, []uint64 /* canIgnore */, []uint64 /* canAccess */, error) {
+	return lr.resolveLocks(bo, callerStartTS, locks, true, lite)
 }
 
-func (lr *LockResolver) resolveLocks(bo *retry.Backoffer, callerStartTS uint64, locks []*Lock, forWrite bool, lite bool) (int64, []uint64 /*pushed*/, error) {
```

Comment:
Without `forWrite` flag, it seems the mechanism for conflict of optimistic transactions is missing.

Reply:
It's handled here https://github.com/tikv/client-go/pull/367.